### DISCUSSION TITLE
Error Collection with Pagination

### DISCUSSION THREAD

**Discussion by ighmouraceneb** [livewire/livewire#3223] [7. Bugs (archived)] [answered]
The question has been asked several times before. But unfortunately there is not yet a definitive solution.

Context

Livewire version: 2.5.1
Laravel version: 8.0
Browser: Chrome, Firefox

![image](https://user-images.githubusercontent.com/17543179/124272443-051ad800-db3f-11eb-8ecc-fb9ebb72f6e7.png)

So far I have added this condition : 
`|| $value instanceof LengthAwarePaginator`

 in function :
`dehydrate($instance, $response)`

![image](https://user-images.githubusercontent.com/17543179/124273410-4790e480-db40-11eb-9d3a-2d1b91014833.png)

the error has disappeared.

what do you think?

**Comment by joshhanley** [comment]
@ighmouraceneb so as you have pointed out, currently there isn't support for properties that are any of the paginator types. 

It might be as simple as adding an option in there, but there is a few things that need to be checked, and that is to ensure that the paginated collection can be serialised correctly and that it does, and also need to check it then gets rehydrated properly back to a paginated collection.

Hope this helps!

**Comment by ousid** [comment, marked as answer]
I did a trick, it may help you out. I used macros to create a new paginate collection method.

In `app/AppServiceProvider` and in the `boot` method, do the following:

```php
use App\Helpers\CollectionHelper;
use Illuminate\Support\Collection;

// ...
public function boot()
{
 // ....

        Collection::macro('paginate', function (Collection $results, $pageSize) { 
            return CollectionHelper::paginate($results, $pageSize);
        });
}
```

After, you need to create a `CollectionHelper` inside `App/Helpers` or whatever location you want

```php

namespace App\Helpers;

use Illuminate\Container\Container;
use Illuminate\Pagination\LengthAwarePaginator;
use Illuminate\Pagination\Paginator;
use Illuminate\Support\Collection;

class CollectionHelper
{
    public static function paginate(Collection $results, $pageSize)
    {
        $page = Paginator::resolveCurrentPage('page');

        $total = $results->count();

        return self::paginator($results->forPage($page, $pageSize), $total, $pageSize, $page, [
            'path' => Paginator::resolveCurrentPath(),
            'pageName' => 'page',
        ]);
    }

    /**
     * Create a new length-aware paginator instance.
     *
     * @param  \Illuminate\Support\Collection  $items
     * @param  int  $total
     * @param  int  $perPage
     * @param  int  $currentPage
     * @param  array  $options
     * @return \Illuminate\Pagination\LengthAwarePaginator
     */
    protected static function paginator($items, $total, $perPage, $currentPage, $options)
    {
        return Container::getInstance()->makeWith(LengthAwarePaginator::class, compact(
            'items',
            'total',
            'perPage',
            'currentPage',
            'options'
        ));
    }
}
```

After all this, you can add a little helper function inside your `helper.php` to make it works inside blade files

```php
use Illuminate\Support\Collection;

if (!function_exists('paginate_collection')) {

    /**
     * paginate collection
     * @param collection $results
     * @param int $pageNumber
     * @return \Illuminate\Pagination\LengthAwarePaginator
     */

    function paginate_collection($results, $pageNumber)
    {
        return Collection::paginate($results, $pageNumber);
    }
}
```

Now, we are done and ready to use your new collection:

Inside your component:
```php
public function render()
{
 return view('....', [
    'users' => User::get(), // it must be a collection instance
  ]);
}
```
And in the blade:
```blade
  @foreach(paginate_collection($users, 10) as $user)
           ....
  @endforeach

   {{ $users->render() }}
```
With this approach should work perfectly! I know it's long and this is a temporary solution till next updates in livewire.

Happy coding.

**Reply by KevinMakwanaBacancy** [reply]
Hello @ousid , I followed all given steps and when I apply  {{ $users->render() }} in blade file, I received below error.

BadMethodCallException
Method Illuminate\Database\Eloquent\Collection::render does not exist.

**Reply by ousid** [reply]
I'm not sure TBH, it's been a while since I used this trick.

What about `paginate_collection($users, 10)->render()`?

**Reply by KevinMakwanaBacancy** [reply]
I resolved it by doing this.

 @php
            $patinatedTickets = paginate_collection($technicianTickets,2);    
 @endphp

@foreach($patinatedTickets as $ticket)
@endforeach

{{ $patinatedTickets->links() }}


It works for me :)

**Reply by ousid** [reply]
Nice! Glade it got solved :)
